client/main.coffee

```diff
@@ -768,6 +768,21 @@ tools =
     hotkey: '0'
     once: ->
       currentBoard().setScale 1
+  pageZoomFit:
+    icon: 'expand-arrows-alt'
+    help: 'Zoom to fit'
+    hotkey: [')', 'Shift-0']
+    once: ->
+      elts =
+        for elt in currentBoard().root.childNodes
+          continue if elt.classList.contains 'highlight'
+          continue if elt.classList.contains 'selected'
+          continue if elt.classList.contains 'outline'
+          continue if elt.classList.contains 'grid'
+          continue unless elt.dataset.id
+          elt
+      {min, max} = dom.unionSvgExtremes currentBoard().svg, elts, currentBoard().root
```

Comment:
I think this code for constructing "real child elements" (also used in export) should be factored out. Maybe a method of `Board`. And maybe worth also adding a `bbox` method which does that plus the `unionSvgExtremes`.

Reply:
I've refactored by adding four helper functions to `Board`:
- `Board.filterChildren(predicate)` returns a list of child elements where `predicate el` is true
- `Board.childrenExcludingClasses(excludeList)` filters down to elements where no members of `excludeList` appear in their `classList`
- `Board.drawnChildren()` returns children whose `classList`s do not contain one of `['highlight', 'selected', 'outline', 'grid']`
- `Board.drawnBBox()` returns the bbox, computed from `dom.unionSvgExtremes`, of the `drawnChildren`

I expect these functions might simplify some other code, but I did not want to mix that into this commit